src/snowflake/cli/_plugins/nativeapp/manager.py

```diff
@@ -490,15 +493,11 @@ def get_existing_app_info(self) -> Optional[dict]:
             )
 
     def get_existing_app_pkg_info(self) -> Optional[dict]:
-        """
-        Check for an existing application package by the same name as in project definition, in account.
-        It executes a 'show application packages like' query and returns the result as single row, if one exists.
-        """
-
-        with self.use_role(self.package_role):
-            return self.show_specific_object(
-                "application packages", self.package_name, name_col=NAME_COL
-            )
+        return ApplicationPackageEntity.get_existing_app_pkg_info(
+            sql_executor=self,
```

Comment:
Do we even need an instance to execute the SQL commands? Would we be able to turn the mixin into a stateless utility instead? Or at least I'd try to extract a contract out of it (SqlExecutor?) and leave the mixin out of the equation (but the mixin implements the contract, clearly).

Reply:
Created `SqlExecutor` and a `get_sql_executor()` to avoid using `SqlExecutionMixin` directly and passing an sql_executor instance.